### ISSUE TITLE
Cannot reproduce recall score for Autoencoder on BGL dataset

### ISSUE THREAD

**Issue by andrzejzera** [logpai/deep-loglizer#8]
It's great you open sourced your work on Deep Learning models for anomaly detection of system logs.

However, I'm struggling to reproduce the result from your paper in one particular case:
- BGL dataset
- w/o semantics
- Autoencoder model

In your paper, you report 0.791 for precision, 0.773 for recall and 0.782 for F1 score while I'm getting consistently higher values for recall with comparable precision (resulting in higher F1 score overall than in the paper).

I'm running the appropriate code from the benchmark folder:
```python demo/ae_demo.py --feature_type sequentials --anomaly_ratio 0.8 --dataset BGL --data_dir BGL/bgl_0.0_tar```

And from the very first epoch, I'm getting recall > 0.9 (in each of the model runs results are slightly different, but almost always recall is higher than 0.9 which is significantly higher than 0.773 in the paper):
```
2021-12-26 23:27:01,317 P198 INFO Start training on 2063 batches with cuda:0.
2021-12-26 23:28:40,466 P198 INFO Epoch 1/100, training loss: 0.00001 [99.15s]
2021-12-26 23:28:40,466 P198 INFO Evaluating test data.
2021-12-26 23:28:50,525 P198 INFO Finish inference [10.06s]
2021-12-26 23:28:51,160 P198 INFO NumExpr defaulting to 2 threads.
2021-12-26 23:28:51,166 P198 INFO {'f1': '0.812', 'rc': '0.932', 'pc': '0.719', 'acc': '0.734'}
2021-12-26 23:28:51,302 P198 INFO Saving model to ./experiment_records/5035e31a/model.ckpt
2021-12-26 23:30:30,547 P198 INFO Epoch 2/100, training loss: 0.00000 [99.24s]
2021-12-26 23:30:30,547 P198 INFO Evaluating test data.
2021-12-26 23:30:41,014 P198 INFO Finish inference [10.47s]
2021-12-26 23:30:41,566 P198 INFO {'f1': '0.842', 'rc': '0.966', 'pc': '0.746', 'acc': '0.776'}
2021-12-26 23:30:41,691 P198 INFO Saving model to ./experiment_records/5035e31a/model.ckpt
2021-12-26 23:32:22,105 P198 INFO Epoch 3/100, training loss: 0.00000 [100.41s]
2021-12-26 23:32:22,105 P198 INFO Evaluating test data.
2021-12-26 23:32:33,478 P198 INFO Finish inference [11.37s]
2021-12-26 23:32:34,020 P198 INFO {'f1': '0.851', 'rc': '0.977', 'pc': '0.754', 'acc': '0.790'}
2021-12-26 23:32:34,153 P198 INFO Saving model to ./experiment_records/5035e31a/model.ckpt
2021-12-26 23:34:14,924 P198 INFO Epoch 4/100, training loss: 0.00000 [100.76s]
2021-12-26 23:34:14,924 P198 INFO Evaluating test data.
2021-12-26 23:34:25,592 P198 INFO Finish inference [10.67s]
2021-12-26 23:34:26,139 P198 INFO {'f1': '0.871', 'rc': '1.000', 'pc': '0.772', 'acc': '0.818'}
2021-12-26 23:34:26,272 P198 INFO Saving model to ./experiment_records/5035e31a/model.ckpt
2021-12-26 23:36:06,985 P198 INFO Epoch 5/100, training loss: 0.00000 [100.71s]
2021-12-26 23:36:06,985 P198 INFO Evaluating test data.
2021-12-26 23:36:17,473 P198 INFO Finish inference [10.49s]
2021-12-26 23:36:18,025 P198 INFO {'f1': '0.851', 'rc': '0.977', 'pc': '0.754', 'acc': '0.790'}
2021-12-26 23:37:59,357 P198 INFO Epoch 6/100, training loss: 0.00000 [101.21s]
2021-12-26 23:37:59,357 P198 INFO Evaluating test data.
2021-12-26 23:38:09,894 P198 INFO Finish inference [10.54s]
2021-12-26 23:38:10,448 P198 INFO {'f1': '0.872', 'rc': '0.966', 'pc': '0.794', 'acc': '0.825'}
2021-12-26 23:38:10,581 P198 INFO Saving model to ./experiment_records/5035e31a/model.ckpt
2021-12-26 23:39:51,641 P198 INFO Epoch 7/100, training loss: 0.00000 [101.05s]
2021-12-26 23:39:51,641 P198 INFO Evaluating test data.
2021-12-26 23:40:02,132 P198 INFO Finish inference [10.49s]
2021-12-26 23:40:02,688 P198 INFO {'f1': '0.851', 'rc': '0.977', 'pc': '0.754', 'acc': '0.790'}
2021-12-26 23:41:41,796 P198 INFO Epoch 8/100, training loss: 0.00000 [98.98s]
2021-12-26 23:41:41,796 P198 INFO Evaluating test data.
2021-12-26 23:41:52,122 P198 INFO Finish inference [10.33s]
2021-12-26 23:41:52,668 P198 INFO {'f1': '0.864', 'rc': '0.977', 'pc': '0.775', 'acc': '0.811'}
2021-12-26 23:43:32,220 P198 INFO Epoch 9/100, training  loss: 0.00000 [99.42s]
2021-12-26 23:43:32,221 P198 INFO Evaluating test data.
2021-12-26 23:43:42,340 P198 INFO Finish inference [10.12s]
2021-12-26 23:43:42,873 P198 INFO {'f1': '0.876', 'rc': '0.966', 'pc': '0.802', 'acc': '0.832'}
2021-12-26 23:43:43,000 P198 INFO Saving model to ./experiment_records/5035e31a/model.ckpt
2021-12-26 23:45:21,095 P198 INFO Epoch 10/100, training loss: 0.00000 [98.09s]
2021-12-26 23:45:21,095 P198 INFO Evaluating test data.
2021-12-26 23:45:31,210 P198 INFO Finish inference [10.11s]
2021-12-26 23:45:31,757 P198 INFO {'f1': '0.762', 'rc': '0.875', 'pc': '0.675', 'acc': '0.664'}
2021-12-26 23:47:09,406 P198 INFO Epoch 11/100, training loss: 0.00000 [97.51s]
2021-12-26 23:47:09,406 P198 INFO Evaluating test data.
2021-12-26 23:47:20,383 P198 INFO Finish inference [10.98s]
2021-12-26 23:47:20,925 P198 INFO {'f1': '0.876', 'rc': '0.966', 'pc': '0.802', 'acc': '0.832'}
2021-12-26 23:48:59,562 P198 INFO Epoch 12/100, training loss: 0.00000 [98.51s]
2021-12-26 23:48:59,562 P198 INFO Evaluating test data.
2021-12-26 23:49:09,891 P198 INFO Finish inference [10.33s]
2021-12-26 23:49:10,432 P198 INFO {'f1': '0.876', 'rc': '0.966', 'pc': '0.802', 'acc': '0.832'}
2021-12-26 23:49:10,565 P198 INFO Early stop at epoch: 12

**Comment by zbchern** [comment]
Hi, it seems that you are using the dataset without anomalies in the training data. Please try

`python demo/ae_demo.py --feature_type sequentials --anomaly_ratio 0.8 --dataset BGL --data_dir BGL/bgl_0.8_tar`
